### ISSUE TITLE
no-multiple-empty-lines should ignore multi-line strings

### ISSUE THREAD

**Issue by ilyavolodin** [eslint/eslint#2624]
We added a fix for `no-multiple-empty-lines` to ignore template strings, but it should also ignore multi-line strings, and probably block comments as well.

**Comment by xjamundx** [comment]
Aren't multiline stings illegal?

**Comment by nzakas** [comment]
```
var x = "foo\
bar";
```

**Comment by gyandeeps** [comment]
I think it ignores multi-line strings.
I ran the test with this and it doesn't throw any errors:

```js
        {
            code: "x = 'hi\\\n \\\n bye\\\n hi';",
            options: [{ max: 0 }]
        }
```

**Comment by lo1tuma** [comment]
@gyandeeps you are right. That’s because the lines aren’t actually empty, they contain at least a backslash.

**Comment by gyandeeps** [comment]
And even this:

```js
       {
            code: "var y = 1;\n /**\n * hi \n*/\nfunction hi(){\n}",
            options: [{ max: 0 }]
        }
``

**Comment by gyandeeps** [comment]
multi-line strings: they have to have a back slash otherwise it would be illegal expression.

**Comment by lo1tuma** [comment]
@gyandeeps yes I know. I think we can close this issue as invalid.

**Comment by gyandeeps** [comment]
how about the block-comment scenario as mentioned above.
Do we need to fix that or it is correct the way its behaving right now?

**Comment by lo1tuma** [comment]
I think it would be reasonable to warn about multiple empty line even within multiline comments. If someone wants an exceptions for multiline comments it should be an additional feature behind an option.

**Comment by gyandeeps** [comment]
@lo1tuma I agree.

**Comment by gyandeeps** [comment]
@ilyavolodin should we close this?

**Comment by ilyavolodin** [comment]
Agree
